src/transform/mod.rs

```diff
@@ -71,6 +71,29 @@ pub enum TxType {
   H_FLIPADST = 15,
 }
 
+impl TxType {
+  /// Compute transform type for inter chroma.
+  ///
+  /// <https://aomediacodec.github.io/av1-spec/#compute-transform-type-function>
+  #[inline]
+  pub fn uv_inter(self: Self, uv_tx_size: TxSize) -> Self {
```

Comment:
Thanks for catching the reason and fixes this!
I indeed missed a step to check the tx_size for each plane as stated in spec. (So, more tx_type is available for uv channels, since its tx_size is smaller after chroma sampling except 444)
I see that there are three cases for uv_tx_size:  1) 32x32, 2) 16x16, 3) all others.
I think 3) is meant for uv_tx_size < 16x16.
What would happen if uv_tx_size = 64x64? This happens for chroma sampling 444, right?
I guess this might explain the failure with your 444 test, if it still fails.

Reply:
As you referred, https://aomediacodec.github.io/av1-spec/#compute-transform-type-function has
"txSzSqrUp > TX_32X32", case which returns DCT_DCT.